### PR TITLE
Windows.UI.Color considered harmful

### PR DESCRIPTION
**Pull request by zadjii-msft**
_targets #192_

Adds our own struct to the API for passing colors around. This is lifted straight out of the Terminal, which had to deal with this quite a lot. 

Since this color is a struct (so that it trivially marshals across the ABI), it can't be _null_. So we also added `OptionalColor`, which is explicitly "Maybe a color, maybe not". We were bit enough times by trying to have a special secret `INVALID_COLOR` value that it ultimately made sense just to put another bool in there. 

While I'm touching the SDK and breaking everyone, I figured I'd also do the `s/Loading/IsLoading/g` thing now too.

### CODE REVIEW

**Comment by michael-hawker** [comment]
QQ: While we're cleaning up names, should it be `HasMoreItems` and `HasDetails` or `IsDetailsVisible`?

**Comment by zadjii-msft** [comment]
note to self, `dev/migrie/f/icon-factory` is branched off here, which means it's now stacked 3 high

**Comment by zadjii-msft** [comment]
> QQ: While we're cleaning up names, should it be `HasMoreItems` 

Yea this makes sense, this I agree with fully.

> and `HasDetails` or `IsDetailsVisible`?

This I'm less sure about. The way I pictured it, not every extension that has details on its items necessarily needs them shown by default. So `ShowDetails=true` was supposed to be a way for an extension to say "please auto-expand details by default". `IsDetailsVisible` feels similar, but also I'm not sure if it's more correct or not (words are hard)

File: src/modules/cmdpal/WindowsCommandPalette/PageViewModel.cs [hunk 1]
```diff
@@ -17,7 +16,20 @@ public class PageViewModel
     // public IPage PageAction { get => pageAction; set => pageAction = value; }
     public ActionViewModel Command { get; }
 
-    public Color AccentColor => PageAction.AccentColor;
+    public Windows.UI.Color AccentColor
```
Comment (michael-hawker) [inline src/modules/cmdpal/WindowsCommandPalette/PageViewModel.cs:19]: Forgot this was the old app for a second... 😅

Comment (zadjii-msft) [reply src/modules/cmdpal/WindowsCommandPalette/PageViewModel.cs:19]: Oh yea that codebase is in FULL ON "yolo as long as it still kinda runs" mode

**Comment by michael-hawker** [comment]
Yeah, I guess let's keep `ShowDetails` for now. We can always get more feedback from folks later.

**Review by michael-hawker — APPROVED** [review]
LGTM

**Pull request merged**